pkg/schema/tracker.go

```diff
@@ -311,75 +306,8 @@ func (tr *Tracker) CreateTableIfNotExists(db, table string, ti *model.TableInfo)
 		return infoschema.ErrDatabaseNotExists.GenWithStackByArgs(dbName)
 	}
 
-	// we need to go through the low-level DDL Job API since we don't have a way
-	// to recover a CreateTableStmt from a TableInfo yet.
-
-	// First enqueue the DDL job.
-	var (
-		jobID int64
-		ctx   = context.Background()
-	)
-	err := kv.RunInNewTxn(ctx, tr.store, true /*retryable*/, func(_ context.Context, txn kv.Transaction) error {
-		// reallocate IDs
-		idsCount := 2
-		if ti.Partition != nil {
-			idsCount += len(ti.Partition.Definitions)
-		}
-		m := meta.NewMeta(txn)
-		ids, err := m.GenGlobalIDs(idsCount)
-		if err != nil {
-			return err
-		}
-
-		jobID = ids[0]
-		tableInfo := cloneTableInfo(ti)
-		tableInfo.ID = ids[1]
-		tableInfo.Name = tableName
-		if tableInfo.Partition != nil {
-			for i := range tableInfo.Partition.Definitions {
-				tableInfo.Partition.Definitions[i].ID = ids[i+2]
-			}
-		}
-
-		return m.EnQueueDDLJob(&model.Job{
-			ID:         jobID,
-			Type:       model.ActionCreateTable,
-			SchemaID:   dbInfo.ID,
-			TableID:    tableInfo.ID,
-			SchemaName: dbName.O,
-			Version:    1,
-			StartTS:    txn.StartTS(),
-			BinlogInfo: &model.HistoryInfo{},
-			Args:       []interface{}{tableInfo},
-		})
-	})
-	if err != nil {
-		return err
-	}
-
-	// Then wait until the DDL job is synchronized (should take 2 * lease)
-	lease := tr.dom.DDL().GetLease() * 2
-	for i := 0; i < waitDDLRetryCount; i++ {
-		var job *model.Job
-		err = kv.RunInNewTxn(ctx, tr.store, false /*retryable*/, func(_ context.Context, txn kv.Transaction) error {
-			m := meta.NewMeta(txn)
-			var e error
-			job, e = m.GetHistoryDDLJob(jobID)
-			return e
-		})
-		if err == nil && job != nil {
-			if job.IsSynced() {
-				return nil
-			}
-			if job.Error != nil {
-				return job.Error
-			}
-		}
-		time.Sleep(lease)
-	}
-	if err == nil {
-		// reaching here is basically a bug.
-		return errors.Errorf("Cannot create table %s.%s, the DDL job never returned", db, table)
-	}
-	return err
+	d := domain.GetDomain(tr.se).DDL()
+	ti = cloneTableInfo(ti)
+	ti.Name = tableName
+	return d.CreateTableWithInfo(tr.se, dbName, ti, ddl.OnExistError, false)
```

Comment:
> ddl.OnExistError

please add a unit test about checking IfNotExists

Reply:
how did IfNotExists happen 😂

Reply:
should not happen，so I use error here